services/ui/ws/display.cc

```diff
@@ -198,7 +198,11 @@ void Display::SetProperty(const std::string& name, const std::vector<uint8_t>* v
   DCHECK(window_server_->IsInExternalWindowMode());
   DCHECK(name == mojom::WindowManager::kShowState_Property);
 
-  const int64_t state = mojo::ConvertTo<int64_t>(*value);
+  auto state = static_cast<ui::mojom::ShowState>(mojo::ConvertTo<int64_t>(*value));
+  if (root_->GetShowState() == state)
+    return;
+
+  root_->SetProperty(name, value);
```

Comment:
What's the point of setting property to the root here? AFAIK, Display::SetProperty is called from visible ServerWindow anyway.